hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/contract/AbstractContractDistCpTest.java

```diff
@@ -628,8 +629,15 @@ public void testDistCpWithIterator() throws Exception {
     GenericTestUtils
         .createFiles(remoteFS, source, getDepth(), getWidth(), getWidth());
 
+    GenericTestUtils.LogCapturer log =
```

Comment:
Not over-enamoured of this test strategy. As well as being brittle, every use of this is going to complicate our move to Log4J2.

Is there any other way we could do this?

If the distcp test collected the IOStatistics of iteration, then those stores which returned it from iterators (s3a currently, abfs is still WiP) could count it and actually assert on real invocations. HDFS Doesn't do this.

Reply:
Thanx @steveloughran, do you suggest that we should have two options like -useiteratorforsource and -useiteratorfortarget. Do you think in that case we would be able to save out on memory? since the target list is being build as part of CopyCommitter, so even if one takes the normal path,  We would get OOM, just `when` will differ?

Regarding the log stuff, That was the only thing I could think of, to confirm if iterator was used. And during migration to Log4J2, Will moving to something like this will also be of no help:
https://github.com/apache/hive/blob/master/ql/src/test/org/apache/hadoop/hive/ql/metadata/TestHive.java#L285

Reply:
* hive code looks great!

Reply:
So, Are you Ok with the other changes? This logger change we can not do now itself in the code, need to do during the migration itself, Was just trying to find a solution to the problem you told, or if there isn't any